flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/TaskIOMetricGroup.java

```diff
@@ -196,6 +204,23 @@ public void markTaskStart() {
         this.taskStartTime = System.currentTimeMillis();
     }
 
+    public void markTaskInitialize() {
+        this.taskInitializeTime = System.currentTimeMillis();
+    }
+
+    /**
+     * Return the duration of one task spends on restoring/initialization, return 0 when the task is
+     * not in initialization.
+     */
+    @VisibleForTesting
+    public long getTaskInitializeTime() {
```

Comment:
I think this metric should be more similar to `TotalTime` or `lastCheckpointRestoreTimestamp`: Return real value even if JobStatus has changed/restored.
WDYT?